# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/WindowNamedPropertiesHandler.cpp
# Commit: 0a44c410b59b
# Full Hash: 0a44c410b59bbc54bfd80ce598562e4362961d7e
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-24 09:51:27
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 20dbcfc9eacc -r 0a44c410b59b dom/base/WindowNamedPropertiesHandler.cpp
--- a/dom/base/WindowNamedPropertiesHandler.cpp	Thu Apr 23 21:52:44 2020 +0000
+++ b/dom/base/WindowNamedPropertiesHandler.cpp	Thu Apr 23 21:52:47 2020 +0000
@@ -169,7 +169,7 @@
   nsGlobalWindowOuter* outer = win->GetOuterWindowInternal();
   if (outer) {
     if (BrowsingContext* bc = outer->GetBrowsingContext()) {
-      for (const auto& child : bc->GetChildren()) {
+      for (const auto& child : bc->Children()) {
         const nsString& name = child->Name();
         if (!name.IsEmpty() && !names.Contains(name)) {
           // Make sure we really would expose it from getOwnPropDescriptor.